File: parcel-locker-cpp/library/src/repositories/ClientRepository.cpp
//
// Created by student on 10.06.2021.
//

#include "repositories/ClientRepository.h"
#include "model/Client.h"

ClientPtr ClientRepository::findByTelNumber(const std::string& telNumber) const {
    for (auto client : objects)
    {
        if(client != nullptr && client->getTelNumber() == telNumber)
        {
            return client;
        }
    }
    return nullptr;
}

File: parcel-locker/src/main/java/pl/pas/parcellocker/model/client/ClientRepository.java
package pl.pas.parcellocker.model.client;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface ClientRepository {
    void add(Client client);

    void archive(UUID clientId);

    List<Client> findAll();

    Optional<Client> findByTelNumber(String telNumber);

    List<Client> findByTelNumberPart(String telNumberPart);
}
